imports/plugins/included/surcharges/server/no-meteor/util/surchargeCheck.js

```diff
@@ -0,0 +1,77 @@
+import { operators, propertyTypes } from "./helpers";
+
+/**
+ * @summary Filter surcharges based on surcharge restriction data
+ * @param {Object} surcharge - surcharge to check attributes and destination against
+ * @param {Object} extendCommonOrder - details about the purchase a user wants to make.
+ * @returns {Bool} true / false as to whether method is still valid after this check
+ */
+export async function surchargeCheck(surcharge, extendCommonOrder) {
+  const { items, shippingAddress } = extendCommonOrder;
+  const { attributes, destination } = surcharge;
+
+  const validSurcharge = items.some((item) => { // eslint-disable-line
+    // For each item, check against surcharge
+    const { country, postal, region } = destination;
```

Comment:
Testing with a surcharge with attributes, but without destination. When I tried adding to cart, there was an error that I think might be from this line. The error:

```
Cannot destructure property `country` of 'undefined' or 'null'.
```

Reply:
Updated: moved the destructuring to after we check if destination exists.